routers/google.controller.js

```diff
@@ -0,0 +1,31 @@
+const { userModel } = require('../database/models');
+
+class GoogleController {
+  constructor() {}
+  async findOrCreate(profile, callback) {
+    try {
+      const { email } = profile._json;
+      const userName = email.substring(0, email.indexOf('@'));
+      const foundUser = await userModel.findOne({ email });
```

Comment:
We could do findOrCreate in one mongoDB request with `findOneAndUpdate + upsert + $setOnInsert`